packages/nimble-components/src/table/models/table-layout-manager.ts

```diff
@@ -222,10 +222,16 @@ export class TableLayoutManager<TData extends TableRecord> {
     }
 
     private resetGridSizedColumns(): void {
-        if (!this.gridSizedColumns) {
+        if (!this.gridSizedColumns?.length) {
```

Comment:
nit / pref: Be explicit on size checks, feel like truthiness on size checks can be easily overlooked

```suggestion
        if (!this.gridSizedColumns || this.gridSizedColumns.length === 0) {
```